src/getPage.tsx

```diff
@@ -54,16 +56,32 @@ export default async function getPage({
   };
   // @TODO: Consider printing extended options value behind a debug flag
 
-  const makePage = async (optionsOverride?: Partial<ExtendedOptions>) => {
+  const makePage = async (
+    optionsOverride?: Partial<ExtendedOptions>
+  ): Promise<{ pageElement: JSX.Element; pageObject: PageObject }> => {
     const mergedOptions = { ...options, ...optionsOverride };
 
     const pageObject = await getPageObject({
       options: mergedOptions,
     });
 
+    const pageData = await fetchRouteData({
+      pageObject,
+      options: mergedOptions,
+    });
+
+    if (pageData.redirect) {
+      return makePage({
+        ...mergedOptions,
+        route: pageData.redirect.destination,
+        isClientSideNavigation: false,
```

Comment:
Is `pageData.redirect` to be always considered a server-side navigation?

Reply:
Yes, this is only supported on `getServerSideProps` and `getStaticProps` -- both are always executed server side.

Reply:
What I referred to is the case of a user navigating client side to a page which triggers a redirect. Next.js doesn't return any 302 but the application performs a seamless client side transition to the final destination route. I was wandering whether this `isClientSideNavigation: false` replicates this behaviour.

Reply:
Yeah, this already works as expected. I have removed this line as it just made confusion and was wrong (however, it did not influence anything in our logic).